core/FoundDefinitions.h

```diff
@@ -196,6 +196,7 @@ struct FoundField {
     };
     Kind kind;
     bool onSingletonClass;
+    bool fromWithinMethod;
```

Comment:
Out of curiosity, how did this work previously when it was done in resolver, not namer?

Reply:
It work(ed/s) correctly in resolver because the `Context` structure that we get has a method as its `owner` field.  We decided to always have fields owned by classes, not methods, so we have to hack things a bit here.

It's very sad that this code is *almost* `methodOwner`...but not quite.